api.ts

```diff
@@ -1222,6 +1225,12 @@ export type MetadataFunction = (
   formulaContext?: MetadataContext,
 ) => Promise<MetadataFormulaResultType | MetadataFormulaResultType[] | ArraySchema | ObjectSchema<any, any>>;
 
+export type CellAutocompleteMetadataFunction = (
+  context: CellAutocompleteExecutionContext,
+  search: string,
+  formulaContext?: MetadataContext,
```

Comment:
Agree with Eric that we probably need the sync table params somewhere in here, for the same reasons that @chris-codaio decided we need them in the update() function. Not sure the clearest interface for that, but should be able to use the type used in the update() function, and either make it a new param here or dangle it off one of the contexts.

Reply:
Seems like the bigger issue is that the sync table params can vary depending on which sync step caused this row to appear, and I don't think we currently have a way to figure that out.

Hypothetically maybe we could add a new column to the source table to track which sync step is responsible for its appearance?...

cc @huayang-codaio

Reply:
yeah I think you need that for 2-way.